src/service/template-impl.js

```diff
@@ -264,8 +264,19 @@ export class Templates {
   findTemplate(parent, opt_querySelector) {
     const templateElement = this.maybeFindTemplate(parent, opt_querySelector);
     user().assert(templateElement, 'Template not found for %s', parent);
-    user().assert(templateElement.tagName == 'TEMPLATE',
-        'Template element must be a "template" tag %s', templateElement);
+    const templateTagName = templateElement.tagName;
+    user().assert(
+        templateTagName == 'TEMPLATE' || templateTagName == 'SCRIPT',
+        'Template must be a "template" tag or defined in a "script" tag %s',
+        templateElement);
+    if (templateTagName == 'SCRIPT') {
+      user().assert(
+          templateElement.getAttribute('type') == 'text/plain',
+          'Script template attribute must be "text/plain"');
+      user().assert(
+          templateElement.getAttribute('template') == 'amp-mustache',
+          'Script template attribute must be "amp-mustache"');
```

Comment:
We shouldn't couple template-impl.js to amp-mustache. Mustache is just a flavor of template.